modules/ingestor/src/graph/vulnerability/mod.rs

```diff
@@ -57,6 +58,10 @@ impl Graph {
         information: I,
         tx: TX,
     ) -> Result<VulnerabilityContext, Error> {
+        self.connection(&tx)
+            .execute_unprepared("LOCK TABLE vulnerability IN EXCLUSIVE MODE;")
```

Comment:
should be enough to define unique constraint - are we observing otherwise (which is just another way of asking if we have a unit test that replicates existing collision) ?